.github/workflows/wasmcloud.yml

```diff
@@ -566,13 +533,24 @@ jobs:
     steps:
     - uses: actions/checkout@v4.1.1
 
+    - name: Extract tag context
+      id: ctx
+      run: |
+        version=${GITHUB_REF_NAME#${{ matrix.crate }}-v}
+        echo "version is ${version}"
+        if [[ $version == *"-"* ]]; then
+          echo "version ${version} is a pre-release"
+          echo "prerelease=true" >> "$GITHUB_OUTPUT"
+        fi
+
     - name: dry-run publish ${{ matrix.crate }} to crates.io
-      run: cargo publish --dry-run
-      working-directory: ./crates/${{ matrix.crate }}
       if: ${{ !startsWith(github.ref, 'refs/tags/') }}
       continue-on-error: ${{ matrix.workspace-dependencies }} # publish may fail due to workspace crates not being published yet
+      run: cargo publish --dry-run
+      working-directory: ./crates/${{ matrix.crate }}
 
     - name: publish ${{ matrix.crate }} to crates.io
+      if: startsWith(github.ref, format('refs/tags/{0}-v', matrix.crate)) && !steps.ctx.outputs.prerelease
```

Comment:
Just noting: I was thrown off at first by the inclusion of the `-` here, since the OCI job includes the `-` in `matrix.prefix`. I see why we can't do that here, I just hope we don't make silly errors between the two in the future 😕

Reply:
well, ideally we should never really have to change this

Reply:
I'm tempted to save this comment 😉